adb/adb_protocol.py

```diff
@@ -298,7 +317,8 @@ def Connect(cls, usb, banner=b'notadb', rsa_keys=None, auth_timeout_ms=100):
           raise InvalidResponseError(
               'Unknown AUTH response: %s %s %s' % (arg0, arg1, banner))
 
-        signed_token = rsa_key.Sign(str(banner))
+        # Do not mangle the banner property here by converting it to a string
+        signed_token = rsa_key.Sign(banner)
```

Comment:
Before this was sending a mangled banner to the device instead of the true string. Removing the string cast fixed this.